YJDH-689 | feat: don't autocomplete youth/employee's personally identifiable fields

karisal-anders opened the pull request:
## Description :sparkles:

### feat: don't autocomplete youth/employee's personally identifiable fields

Shared frontend code:
 - TextInput component
   - add autoComplete as input property and use it
 - SocialSecurityNumberInput component
   - force autoComplete="off" for this input as social security numbers
	 should not be auto-completed for security reasons

Kesäseteli frontend code:
 - turn off autocomplete for the following fields:
   - employer UI > employment accordion:
	 - employee_name
	 - summer_voucher_serial_number
	 - employee_ssn (i.e. employee's social security number)
	 - employee_postcode
	 - employee_phone_number
	 - employment_description
   - youth UI > application form:
	 - first_name
	 - last_name
	 - social_security_number
	 - postcode
	 - phone_number
	 - email
   - youth UI > additional info form:
	 - additional_info_description

refs YJDH-689

### feat(fetch_employee_data): add specific "Not found" error message

also make the generic error message clearer in Finnish

refs YJDH-689 (noticed the error message was generic)

## Issues :bug:

[YJDH-689](https://helsinkisolutionoffice.atlassian.net/browse/YJDH-689)

## Testing :alembic:

## Screenshots :camera_flash:

## Additional notes :spiral_notepad:


[YJDH-689]: https://helsinkisolutionoffice.atlassian.net/browse/YJDH-689?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

terovirtanen commented:
**API branch is deployed to platta: https://tet-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**API branch is deployed to platta: https://helsinkilisa-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tet-admin-ui-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**API branch is deployed to platta: https://kesaseteli-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**API branch is deployed to platta: https://tet-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://kesaseteli-employer-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://kesaseteli-handler-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders commented:
@sirtawast, @rikuke, @mjturt I asked for Helsinki benefit team's code review because some shared code changes were included. Specifically adding `autoComplete` as props to `<TextInput>` and always turning off autocomplete in `<SocialSecurityNumberInput>`

terovirtanen commented:
## TestCafe result is __success__ for https://tet-youth-ui-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://kesaseteli-youth-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**API branch is deployed to platta: https://kesaseteli-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**API branch is deployed to platta: https://tet-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**API branch is deployed to platta: https://helsinkilisa-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tet-admin-ui-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tet-youth-ui-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders commented:
Rebased on most recent main, added autoComplete="off" to Kesäseteli's youth application form's form element also, added a more specific error message to Kesäseteli's employer's UI when `fetch_employee_data` endpoint data fetching returns 404 to mean that no data was found with the given information.

terovirtanen commented:
**API branch is deployed to platta: https://kesaseteli-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**API branch is deployed to platta: https://tet-pr2767.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

terovirtanen commented:
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://kesaseteli-employer-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tet-admin-ui-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://kesaseteli-handler-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tet-youth-ui-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2767.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://kesaseteli-youth-pr2767.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela commented on frontend/kesaseteli/employer/src/components/application/form/TextInput.tsx:
```diff
@@ -16,6 +16,7 @@ export type TextInputProps = {
   placeholder?: string;
   helperFormat?: string;
   onChange?: (value: string) => void;
+  autoComplete?: string;
```
wouldn't it be better set the actual available values there? `"on"|"off"|""`

nikomakela replied:
Oh, there are actually quite a lot of those alternatives as shown here https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete. Could the type be linked from React or HDS or somewhere?

nikomakela replied:
I checked from the React `index.d.ts` and it seems to be a `string` there as well!

- [InputHTMLAttributes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L2364)
- [FormHTMLAttributes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L2282)

nikomakela commented on frontend/shared/src/types/input-props.d.ts:
```diff
@@ -10,6 +10,7 @@ type InputProps<T, V = string> = {
   errorText?: string;
   placeholder?: string;
   disabled?: boolean;
+  autoComplete?: string;
```
wouldn't it be better set the actual available values there? "on"|"off"|""

nikomakela reviewed (approved):
The form level configuration should do the trick already, but it does not harm to have those autocomplete setups in each text field too. And it is good to allow the autocomplete features in employer fields and disable those in employee fields. Good job!

karisal-anders commented:
As the shared code changes have not changed since all Helsinki benefit and TET tests passed in this PR (1), the shared code changes are small and look safe enough to me (2) and the PR was approved by @nikomakela (3) (Points 1, 2 and 3 are for @sirtawast, @rikuke & @EmiliaMakelaVincit), and that the autoComplete is typed as string by React also (This point is for @nikomakela) I am merging this now as is. Sorry if one or more of you feel like you've been ignored, but I don't want to wait for more time with this PR now—feel free to contact me and we can discuss anything related to this PR or its handling, thank you!
